tests/unit/type/zset.tcl

```diff
@@ -1552,6 +1739,108 @@ start_server {tags {"zset"}} {
+        assert_error "ERR wrong number of arguments*" {r bzmpop}
+        assert_error "ERR wrong number of arguments*" {r bzmpop 0 1}
+        assert_error "ERR wrong number of arguments*" {r bzmpop 0 1 myzset{t}}
+
+        assert_error "ERR numkeys*" {r bzmpop 1 0 myzset{t} MIN}
+        assert_error "ERR numkeys*" {r bzmpop 1 a myzset{t} MIN}
+        assert_error "ERR numkeys*" {r bzmpop 1 -1 myzset{t} MAX}
+
+        assert_error "ERR syntax error*" {r bzmpop 1 1 myzset{t} bad_where}
+        assert_error "ERR syntax error*" {r bzmpop 1 1 myzset{t} MIN bar_arg}
+        assert_error "ERR syntax error*" {r bzmpop 1 1 myzset{t} MAX MIN}
+        assert_error "ERR syntax error*" {r bzmpop 1 1 myzset{t} COUNT}
+        assert_error "ERR syntax error*" {r bzmpop 1 2 myzset{t} myzset2{t} bad_arg}
+
+        assert_error "ERR count*" {r bzmpop 1 1 myzset{t} MIN COUNT 0}
+        assert_error "ERR count*" {r bzmpop 1 1 myzset{t} MAX COUNT a}
+        assert_error "ERR count*" {r bzmpop 1 1 myzset{t} MIN COUNT -1}
+        assert_error "ERR count*" {r bzmpop 1 2 myzset{t} myzset2{t} MAX COUNT -1}
+    }
+
+    test "BZMPOP with multiple blocked clients" {
+        set rd1 [redis_deferring_client]
+        set rd2 [redis_deferring_client]
+        set rd3 [redis_deferring_client]
+        set rd4 [redis_deferring_client]
+        r del myzset{t} myzset2{t}
+
+        $rd1 bzmpop 0 2 myzset{t} myzset2{t} min count 1
+        $rd2 bzmpop 0 2 myzset{t} myzset2{t} max count 10
+        $rd3 bzmpop 0 2 myzset{t} myzset2{t} min count 10
+        $rd4 bzmpop 0 2 myzset{t} myzset2{t} max count 1
+        wait_for_blocked_clients_count 4
+
+        r multi
+        r zadd myzset{t} 1 a 2 b 3 c 4 d 5 e
+        r zadd myzset2{t} 1 a 2 b 3 c 4 d 5 e
+        r exec
+
+        assert_equal {myzset{t} {{a 1}}} [$rd1 read]
+        assert_equal {myzset{t} {{e 5} {d 4} {c 3} {b 2}}} [$rd2 read]
+        assert_equal {myzset2{t} {{a 1} {b 2} {c 3} {d 4} {e 5}}} [$rd3 read]
+
+        r zadd myzset2{t} 1 a 2 b 3 c
+        assert_equal {myzset2{t} {{c 3}}} [$rd4 read]
+
+        r del myzset{t} myzset2{t}
+    }
+
+    test "BZMPOP propagate as pop with count command to replica" {
+        set rd [redis_deferring_client]
+        set repl [attach_to_replication_stream]
+
+        # BZMPOP without block.
+        r zadd myzset{t} 1 one 2 tow 3 three
+        r zadd myzset2{t} 4 four 5 five 6 six
+        r BZMPOP 0 1 myzset{t} min
+        r BZMPOP 0 2 myzset{t} myzset2{t} max count 10
+        r BZMPOP 0 2 myzset{t} myzset2{t} max count 10
+
+        # BZMPOP with block.
+        $rd BZMPOP 0 1 myzset{t} min count 1
+        wait_for_blocked_client
+        r zadd myzset{t} 1 one
+        $rd BZMPOP 0 2 myzset{t} myzset2{t} min count 5
+        wait_for_blocked_client
+        r zadd myzset{t} 1 one 2 tow 3 three
+        $rd BZMPOP 0 2 myzset{t} myzset2{t} max count 10
+        wait_for_blocked_client
+        r zadd myzset2{t} 4 four 5 five 6 six
```

Comment:
maybe add one that get's released on timeout?
(you'll need to add something else (like `SET x x`) after it and validate it (so we know it didn't propagate anything)

Reply:
Sure, already added